graphql/resolver.go

```diff
@@ -35,6 +36,46 @@ func New(apiURL string) Config {
 			sc: dbConnector,
 		},
 	}
+	c.Directives.RequireHostAccess = func(ctx context.Context, obj interface{}, next graphql.Resolver, access HostAccessLevel) (interface{}, error) {
+		args, isStringMap := obj.(map[string]interface{})
+		if !isStringMap {
+			return nil, ResourceNotFound.Send(ctx, "host not specified")
+		}
+		hostId, hasHostId := args["hostId"].(string)
+		hostIdsInterface, hasHostIds := args["hostIds"].([]interface{})
+		hostIds := []string{}
+		if hasHostIds {
+			for _, v := range hostIdsInterface {
+				hostIds = append(hostIds, v.(string))
+			}
+		}
+		if !hasHostId && !hasHostIds {
+			return nil, ResourceNotFound.Send(ctx, "host not specified")
+		}
+		hostIdsToCheck := []string{}
+		hostIdsToCheck = append(hostIdsToCheck, hostId)
+		hostIdsToCheck = append(hostIdsToCheck, hostIds...)
+		var requiredLevel int
+		if access == HostAccessLevelEdit {
+			requiredLevel = evergreen.HostsEdit.Value
+		} else if access == HostAccessLevelView {
```

Comment:
is access not being specified a valid case? If not I would use `else` rather than `else if`